### ISSUE TITLE
Mallformed URI's leads to crash with `/thing/:id` routes

### ISSUE THREAD

**Issue by jakerdy** [tinyhttp/tinyhttp#399]
**Describe the bug**

I has been testing tinyhttp for my project, and were curious if i could exploit it with `../`-like url's.
And at some point I break it when made a request with url:
```
http://localhost:7331/api/pics/..%2..%2Fmain.yml
```
I got this error: 
```
file:///D:/dev/my_proj/backend/node_modules/@tinyhttp/url/dist/index.js:8
        params[keys[i]] = decodeURIComponent(matches[i + 1]);
                          ^

URIError: URI malformed
    at decodeURIComponent (<anonymous>)
    at getURLParams (file:///D:/dev/my_proj/backend/node_modules/@tinyhttp/url/dist/index.js:8:27)
    at <anonymous> (D:\dev\my_proj\backend\node_modules\@tinyhttp\app\src\app.ts:336:30)
    at loop (D:\dev\my_proj\backend\node_modules\@tinyhttp\app\src\app.ts:357:81)
    at next (D:\dev\my_proj\backend\node_modules\@tinyhttp\app\src\app.ts:359:67)
    at file:///D:/dev/my_proj/backend/node_modules/@tinyhttp/cors/dist/index.js:58:57
    at <anonymous> (D:\dev\my_proj\backend\node_modules\@tinyhttp\app\src\app.ts:27:16)
    at <anonymous> (D:\dev\my_proj\backend\node_modules\@tinyhttp\app\src\app.ts:352:75)
    at loop (D:\dev\my_proj\backend\node_modules\@tinyhttp\app\src\app.ts:357:81)
    at next (D:\dev\my_proj\backend\node_modules\@tinyhttp\app\src\app.ts:359:67)

Node.js v18.17.0
```

Looks like it got broken when url contains two or more `..%2` inside url. Or some other weird urls like that.

**To Reproduce**

1. Create tinyhttp app
2. add handler to route "/api/pics/:id"
3. send request to server at path `/api/pics/..%2..%2Fmain.yml`
4. You will get this error

**Expected behavior**

There are plenty of ways of to hanlndle this situations: 
- parse it and pass as is (unsafe but "fair")
- send `400` responce when parsing failed because it is a `bed request`
- provide some "guard" middleware before routes that will catch bad URIs and allows user to handle such situations.
 
Or something else, it shouldn't be that easy to crash whole server, imo. 

**Versions**

- `node`: v18.17.0
- `@tinyhttp/app`: 2.1.3

**Comment by talentlessguy** [comment]
Seems like a big with `getURLParams` in `packages/url`. We should throw error 400 indeed. PR with fix is welcomr

**Comment by talentlessguy** [comment]
Fixed in in `@tinyhttp/app@2.1.4`
